forum/projects/serializers.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework import serializers
+from .models import Project, Description
+
+
+class DescriptionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Description
+        fields = ['description']
+
+
+class ProjectSerializer(serializers.ModelSerializer):
+    """Nested serializer that handles both Project and its description"""
+
+    description = DescriptionSerializer()
+
+    class Meta:
+        model = Project
+        fields = '__all__'
+        read_only_fields = ['startup', 'is_completed']
+
+    def create(self, validated_data):
+        """Overridden method to create a project with description"""
+        description_data = validated_data.pop('description', None)
+        project = Project.objects.create(**validated_data)
+
+        # Handle description data if data exists
+        if description_data:
+            Description.objects.create(project=project, **description_data)
```

Comment:
Ensure that description data meets specific criteria (e.g., length constraints, non-emptiness) if required.

Reply:
Description serializer takes care of this